orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/api/manifests/helm/HelmBakeManifestRequest.java

```diff
@@ -28,6 +32,24 @@
   @JsonProperty("namespace")
   String namespace;
 
-  @JsonProperty("outputArtifactName")
-  String outputArtifactName;
+  @JsonProperty("overrides")
+  Map<String, Object> overrides;
+
+  @JsonProperty("inputArtifacts")
+  List<Artifact> inputArtifacts;
+
+  List<Artifact> values;
+
+  public HelmBakeManifestRequest(
+      BakeManifestContext bakeManifestContext,
+      List<Artifact> inputArtifacts,
+      String outputArtifactName,
+      Map<String, Object> overrides) {
+    this.setTemplateRenderer(bakeManifestContext.getTemplateRenderer());
```

Comment:
In general the constructor should just directly set properties of the object rather than call setters.  (I think the general reason is that we want to avoid calling methods on a not-yet-fully-initialized class.)

I also think that it would enforce more separation of concerns to call `super(...)` to use a constructor on the superclass for the properties that live there rather than directly setting them here.

Reply:
@ezimanyi I'm not sure what you mean in this case? Each of these classes extend the `BakeManifestRequest` but their inputs are different such that I don't want to have a matching constructor on the super class. It would require passing in more data than needed. Thoughts?